### PR TITLE
RPRBLND-1997: Improve Matlib progress + status update

### PR DESCRIPTION
**Pull request by VascoPi**
### PURPOSE
Improve status update, progress. Handle connection error.

### EFFECT OF CHANGE
Added refresh button. Improved status. Update sync and download progress. Support connection error. 

### TECHNICAL STEPS
Added function update_ui.
Added operator HDUSD_MATLIB_OP_sync_materials.
Added handle connection error.
Added set_status, is_connection methods for Manager.

### CODE REVIEW

File: src/hdusd/ui/matlib.py [hunk 1]
```diff
@@ -109,17 +122,26 @@ def draw(self, context):
         if matlib_prop.search:
             row.operator(HDUSD_MATERIAL_OP_matlib_clear_search.bl_idname, icon='X')
 
+        materials = matlib_prop.get_materials()
+        if manager.poll:
+            manager.status = "No materials found" if not materials else f"{str(len(materials))} materials"
```
Comment (bnagirniak) [inline src/hdusd/ui/matlib.py:127]: f"{len(materials)} materials"

File: src/hdusd/utils/matlib.py [hunk 2]
```diff
@@ -31,6 +33,16 @@
 
 URL = "https://matlibapi.stvcis.com/api"
 MATLIB_DIR = LIBS_DIR.parent / "matlib"
+TIMEOUT = 5
+
+
+def update_ui(area_type="PROPERTIES", region_type="WINDOW"):
```
Comment (bnagirniak) [inline src/hdusd/utils/matlib.py:39]: Maybe move this tool to utils\init?

File: src/hdusd/utils/matlib.py [hunk 3]
```diff
@@ -40,16 +52,23 @@ def download_file(url, path, cache_check=True):
...
+
+        return None
 
     log("download_file", "done")
     return path
 
 
 def download_file_callback(url, path, update_callback, cache_check=True):
-    if cache_check and path.is_file():
+    if cache_check and path.is_file() and path.suffix != ".raw":
```
Comment (bnagirniak) [inline src/hdusd/utils/matlib.py:71]: remove `path.suffix != ".raw"`

File: src/hdusd/utils/matlib.py [hunk 4]
```diff
@@ -40,16 +52,23 @@ def download_file(url, path, cache_check=True):
...
-    with requests.get(url, stream=True) as response:
-        with open(path, 'wb') as f:
-            shutil.copyfileobj(response.raw, f)
+
+    try:
+        with requests.get(url, stream=True, timeout = TIMEOUT) as response:
+            with open(path, 'wb') as f:
+                shutil.copyfileobj(response.raw, f)
+
+    except (requests.ConnectionError, requests.Timeout) as exception:
```
Comment (bnagirniak) [inline src/hdusd/utils/matlib.py:61]: Let's catch exception on top level inside threadpool workers.

**Review by bnagirniak — CHANGES_REQUESTED** [review]
UI changes discussed offline. Move  try...except blocks to higher level.

**Comment by VascoPi** [comment]
Moved to [https://github.com/GPUOpen-LibrariesAndSDKs/BlenderUSDHydraAddon/pull/161](https://github.com/GPUOpen-LibrariesAndSDKs/BlenderUSDHydraAddon/pull/161)

**Pull request closed without merging**